saleor/order/calculations.py

```diff
@@ -0,0 +1,318 @@
+from decimal import Decimal
+from typing import Iterable, Optional, Tuple
+
+from django.conf import settings
+from django.db import transaction
+from django.db.models import prefetch_related_objects
+from django.utils import timezone
+from prices import Money, TaxedMoney
+
+from ..core.prices import quantize_price
+from ..core.taxes import TaxData, TaxError, zero_taxed_money
+from ..discount import OrderDiscountType
+from ..plugins.manager import PluginsManager
+from . import ORDER_EDITABLE_STATUS, utils
+from .interface import OrderTaxedPricesData
+from .models import Order, OrderLine
+
+
+def _recalculate_order_prices(
+    manager: PluginsManager, order: Order, lines: Iterable[OrderLine]
+) -> None:
+    """Fetch taxes from plugins and recalculate order/lines prices.
+
+    Does not throw TaxError.
+    """
+    currency = order.currency
+
+    try:
+        subtotal = zero_taxed_money(currency)
+        for line in lines:
+            variant = line.variant
+            if not variant:
+                subtotal += line.total_price
+                continue
+            product = variant.product
+
+            line_unit = manager.calculate_order_line_unit(order, line, variant, product)
+            line.undiscounted_unit_price = line_unit.undiscounted_price
+            line.unit_price = line_unit.price_with_discounts
+
+            line_total = manager.calculate_order_line_total(
+                order, line, variant, product
+            )
+            line.undiscounted_total_price = line_total.undiscounted_price
+            line.total_price = line_total.price_with_discounts
+
+            line.tax_rate = manager.get_order_line_tax_rate(
+                order, product, variant, None, line_unit.undiscounted_price
+            )
+
+            subtotal += line_total.price_with_discounts
+
+        order.shipping_price = manager.calculate_order_shipping(order)
+        order.shipping_tax_rate = manager.get_order_shipping_tax_rate(
+            order, order.shipping_price
+        )
+        order.total = order.shipping_price + subtotal
+
+    except TaxError:
```

Comment:
I am not sure of this. We will break tax calculations even if a single line will return an exception.  What values will be placed under these fields in case of calling this function for the newly created order?

Reply:
I've changed exception handling to be more granular, so that `order.total` is always calculated.